src/FamilyService.Business/Commands/Child/FindChildrenCommand.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Net;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Kernel.Constants;
+using LT.DigitalOffice.Kernel.Extensions;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.FamilyService.Broker.Requests.Interfaces;
+using LT.DigitalOffice.FamilyService.Business.Commands.Child.Interfaces;
+using LT.DigitalOffice.FamilyService.Data.Interfaces;
+using LT.DigitalOffice.FamilyService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.FamilyService.Models.Db;
+using LT.DigitalOffice.FamilyService.Models.Dto.Models;
+using LT.DigitalOffice.FamilyService.Models.Dto.Requests.Child.Filters;
+
+namespace LT.DigitalOffice.FamilyService.Business.Commands.Child
+{
+  public class FindChildrenCommand : IFindChildrenCommand
+  {
+    private readonly IChildRepository _childRepository;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IResponseCreator _responseCreator;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IChildInfoMapper _childInfoMapper;
+    private readonly IDepartmentService _departmentService;
+
+    public FindChildrenCommand(
+      IChildRepository childRepository,
+      IHttpContextAccessor httpContextAccessor,
+      IResponseCreator responseCreator,
+      IAccessValidator accessValidator,
+      IChildInfoMapper childInfoMapper,
+      IDepartmentService departmentService)
+    {
+      _childRepository = childRepository;
+      _httpContextAccessor = httpContextAccessor;
+      _responseCreator = responseCreator;
+      _accessValidator = accessValidator;
+      _childInfoMapper = childInfoMapper;
+      _departmentService = departmentService;
+    }
+
+    public async Task<FindResultResponse<ChildInfo>> ExecuteAsync(FindChildrenFilter filter)
+    {
+      if (!(await _accessValidator.HasRightsAsync(Rights.AddEditRemoveUsers)
+        || _httpContextAccessor.HttpContext.GetUserId() == filter.ParentUserId))
+      {
+        return _responseCreator.CreateFailureFindResponse<ChildInfo>(HttpStatusCode.Forbidden);
+      }
+
+      FindResultResponse<ChildInfo> response = new();
```

Comment:
What is this for? You return another instance

Reply:
This is for errors tracking(line 60).

Reply:
How are the errors tracking? You collect this and then...?

Reply:
And then I just pass them in the response.

Reply:
Which line?

Reply:
There  is "}" on line 68 and on 65 line you return another instance, so errors will be null anyway